src/tests/sqlsmith/src/expr.rs

```diff
@@ -31,35 +34,63 @@ lazy_static::lazy_static! {
     };
 }
 
+lazy_static::lazy_static! {
+    static ref AGG_FUNC_TABLE: HashMap<DataTypeName, Vec<AggFuncSign>> = {
+        init_agg_table()
+    };
+}
+
 fn init_op_table() -> HashMap<DataTypeName, Vec<FuncSign>> {
     let mut funcs = HashMap::<DataTypeName, Vec<FuncSign>>::new();
     func_sigs().for_each(|func| funcs.entry(func.ret_type).or_default().push(func.clone()));
     funcs
 }
 
+fn init_agg_table() -> HashMap<DataTypeName, Vec<AggFuncSign>> {
+    let mut funcs = HashMap::<DataTypeName, Vec<AggFuncSign>>::new();
+    agg_func_sigs().for_each(|func| funcs.entry(func.ret_type).or_default().push(func.clone()));
+    funcs
+}
+
 impl<'a, R: Rng> SqlGenerator<'a, R> {
-    pub(crate) fn gen_expr(&mut self, typ: DataTypeName) -> Expr {
+    pub(crate) fn gen_expr(&mut self, typ: DataTypeName, can_agg: bool, inside_agg: bool) -> Expr {
         if !self.can_recurse() {
             // Stop recursion with a simple scalar or column.
             return match self.rng.gen_bool(0.5) {
                 true => self.gen_simple_scalar(typ),
-                false => self.gen_col(typ),
+                false => self.gen_col(typ, inside_agg),
             };
         }
-        match self.rng.gen_range(0..=99) {
-            0..=99 => self.gen_func(typ),
+        assert!(!(!can_agg & inside_agg));
+
+        let range = if can_agg & !inside_agg { 99 } else { 90 };
+
+        match self.rng.gen_range(0..=range) {
+            0..=90 => self.gen_func(typ, can_agg, inside_agg),
+            91..=99 => self.gen_agg(typ),
             // TODO: There are more that are not in the functions table, e.g. CAST.
             // We will separately generate them.
             _ => unreachable!(),
         }
     }
 
-    fn gen_col(&mut self, typ: DataTypeName) -> Expr {
-        if self.bound_columns.is_empty() {
-            return self.gen_simple_scalar(typ);
-        }
-        let matched_cols = self
-            .bound_columns
+    fn gen_col(&mut self, typ: DataTypeName, inside_agg: bool) -> Expr {
+        let columns = if inside_agg {
+            if self.bound_relations.is_empty() {
+                return self.gen_simple_scalar(typ);
+            }
+            self.bound_relations
+                .choose(self.rng)
+                .unwrap()
+                .get_qualified_columns()
```

Comment:
How does this differ from `self.bound_columns`?

Reply:
ok This is the things I notice when I use bound_columns. 

For eg, 
Relation t (x int, y int).

Bound_columns will show x and y 
but get_qualified_columns will show t.x and t.y
and this is much more correct, in some case it show ambigious mistake when the query become a bit complex.
so using this is more correct.

Reply:
Hmm seems like a bug. `bound_columns` should be qualified too. Maybe share the failing queries?

Additionally `bound_columns` is required, since it filters for GROUP BY clauses as well.
Using columns from bound_relations does not have this property.

Reply:
Alright sorry I just notice you refer to bound_coloumns.

if you notice, my gen_col function, I did pass a inside_agg to the function.
What this shown is 

Case 1: if currently this expression is inside an aggregate function, (inside_agg is true)
then Whatever coloumns in any of the qualified table should works (even the bounded coloumns restricted by the group by or not. So instead of using boudned coloumns, I call the table (which is originally before you implement the group by). But the difference I use is get_qualified_coloumns() instead of .coloumns because of the issue mentioned above.
Very sorry for the misunderstand of your question .
```
create table t (x int, y int)
select SUM(x), SUM(y) from t group by x
```
These are valid one. so using boudned coloumns would not generate cast like SUM(x)


Case 2: If currently this expression is not inside an aggregate function (inside_agg is false)
then we can only choose the bounded_coloumns restricted by group by (or may not be restricted by group by also).
Valid one
```
select x from t group by x; 
```
Not Valid one
```
select y from t group by x; 
```

I will make proper docs there. Sorry a lot of the confusion

Reply:
Another thing I would like to bring up is, I have been thinking throughout to cover most of generated cases ( May miss lah) and as well as ensuring correctness. And This end up, I put insdie_agg and can_agg (two extra variable), so not only it generate correct expression (like this is why i use bound_relations and sometimes use bound_columns), and by doing so, It is able to generate many complex queries like or even more.
```
select x + SUM(y) from t group by x
```
```
gen_expr( can_agg=true, inside_agg = false)
        Call gen_func() -> binary Add 
                left:
                     gen_expr(can_agg =true, inside_agg = false)
                                gen_col( inside_agg = false) -> Choose from bounded columns
                right:
                      gen_expr(can_agg = true, inside_agg= false)
                                gen_agg() -> Choose SUM()
                                           gen_expr(can_agg = true, inside_agg = true)
                                                     gen_col( inside_agg = true) -> choose from bounded relation.
                                        
```